modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -35,6 +35,27 @@ class LsfExecutor extends AbstractGridExecutor {
 
     private static char BLANK = ' ' as char
 
+    // parse the memory units from $LSF_ENVDIR/lsf.conf
+    def getLSF_memoryUnit() {
+        // assuming $LSF_ENVDIR/lsf.conf exists
+        def lsf_config = new File(System.getenv("$LSF_ENVDIR").toString() + "/lsf.conf")   // should write check to throw informative error if $$LSF_ENVDIR/lsf.conf not found
+        def lines = file1.readLines()
+        def found = lines.find{ line-> line =~ /LSF_UNIT_FOR_LIMITS/ }
+        def memory_units = found.split('LSF_UNIT_FOR_LIMITS=')[1]
+        return memory_units
```

Comment:
Too fragile, I would propose something like: 

```groovy 
def envdir = System.getenv('LSF_ENVDIR')
if( !envdir ) return null
def envfile = new File(envdir,'lsf.conf')
if( !envfile.exists()) return null
for( def line : envfile.readLines() ) {
  if( line.startsWith('LSF_UNIT_FOR_LIMITS=') {
    .. the remaining stuff 
  }
}

Reply:
Much more stable, I used this snippet, thanks!